swaybar/input.c

```diff
@@ -55,32 +55,40 @@ static enum x11_button wl_axis_to_x11_button(uint32_t axis, wl_fixed_t value) {
 	}
 }
 
+void update_cursor(struct swaybar *bar) {
+	struct swaybar_pointer *pointer = &bar->pointer;
+	if (pointer->cursor_theme) {
+		wl_cursor_theme_destroy(pointer->cursor_theme);
+	}
+	int scale = pointer->current ? pointer->current->scale : 1;
+	pointer->cursor_theme = wl_cursor_theme_load(NULL, 24 * scale, bar->shm);
+	struct wl_cursor *cursor;
+	cursor = wl_cursor_theme_get_cursor(pointer->cursor_theme, "left_ptr");
+	pointer->cursor_image = cursor->images[0];
+	wl_surface_set_buffer_scale(pointer->cursor_surface, scale);
+	wl_surface_attach(pointer->cursor_surface,
```

Comment:
It'd be better to damage the whole surface with `wl_surface_damage_buffer` since we're updating it.

Reply:
Do I just need to add a call to `wl_surface_damage_bufer` or can you please explain what I should be changing?

Reply:
Ah, we probably need to bump `wl_compositor` version to 4.